### ISSUE TITLE
Handle keys with initial exclamation mark

### ISSUE THREAD

**Issue by allanlewis** [adrienverge/yamllint#49]
[Jenkins Job Builder][1] supports [custom tags][2] for including external files; these tags all begin `!include`. `yamllint` (incorrectly?) reports an indentation error for source like this:

```yaml
# /tmp/foobar.yaml
---
foo:
  - foobar
  - barbaz

---
!foo:
  - foobar
  - barbaz

```
```
$ yamllint /tmp/foobar.yaml
/tmp/foobar.yaml
  8:3       error    wrong indentation: expected 0 but found 2  (indentation)
```

`yamllint` should treat these keys just like any other, as the YAML spec doesn't appear to exclude exclamation marks from key names - please correct me if I'm wrong there!

[1]: https://docs.openstack.org/infra/jenkins-job-builder/
[2]: https://docs.openstack.org/infra/jenkins-job-builder/definition.html#raw-config

**Comment by adrienverge** [comment]
Hi @allanlewis,

Exclamation points (`!`) are not a valid in key names because [according to the spec](http://yaml.org/spec/1.2/spec.html#!%20tag%20indicator//), `!` defines a *local tag* in YAML.

You can see how `pyyaml` decodes your file:
```yaml
---
!foo:
  - foobar
  - barbaz
```
```shell
python3 -c 'import yaml; print(list(yaml.parse(open("/tmp/foobar.yaml").read())))'
```
```
[StreamStartEvent(),
 DocumentStartEvent(),
 SequenceStartEvent(anchor=None, tag='!foo:', implicit=False),
 ScalarEvent(anchor=None, tag=None, implicit=(True, False), value='foobar'),
 ScalarEvent(anchor=None, tag=None, implicit=(True, False), value='barbaz'),
 SequenceEndEvent(),
 DocumentEndEvent(),
 StreamEndEvent()]
```

This means that it is decoded equivalently to a `[foobar, barbaz]` list, that is locally-tagged `!foo` (and is over-indented by 2 spaces).

Hope this helps!

**Comment by allanlewis** [comment]
Thanks, @adrienverge! So I guess JJB should comply with the spec; never mind.
